src/services/TechRecordsService.ts

```diff
@@ -373,6 +418,55 @@ class TechRecordsService {
         }
     return new HTTPResponse(200, this.formatTechRecordItemForResponse(updatedTechRecord.Attributes as ITechRecordWrapper));
   }
+
+  public addProvisionalTechRecord(techRecord: ITechRecordWrapper, msUserDetails: IMsUserDetails) {
```

Comment:
What do you think about using async instead of nesting promises?
It would make the code more linear, easier to read perhaps and catch sync/async errors
Thoughts?

Reply:
would be different from the other functions as of now. All have similar strucutre.